go-controller/pkg/network-attach-def-controller/network_attach_def_controller.go

```diff
@@ -280,3 +331,28 @@ func nadNeedsUpdate(oldNAD, newNAD *nettypes.NetworkAttachmentDefinition) bool {
 
 	return !reflect.DeepEqual(oldNAD.Spec, newNAD.Spec)
 }
+
+func (nadController *NetAttachDefinitionController) GetActiveNetworkForNamespace(namespace string) (util.NetInfo, error) {
+	if !util.IsNetworkSegmentationSupportEnabled() {
+		return &util.DefaultNetInfo{}, nil
+	}
+	nadController.primaryNetworks.LockKey(namespace)
+	defer nadController.primaryNetworks.UnlockKey(namespace)
+	primaryNetworks, _ := nadController.primaryNetworks.Load(namespace)
+	if len(primaryNetworks) > 1 {
+		return nil, util.NewUnknownActiveNetworkError(namespace)
+	} else if len(primaryNetworks) == 1 {
+		for _, netInfo := range primaryNetworks {
+			// only 1 value
+			return netInfo, nil
+		}
+	}
+
+	return &util.DefaultNetInfo{}, nil
+
+}
+
+// SetPrimaryNetworksForTest is used for testing purposes only
+func (nadController *NetAttachDefinitionController) SetPrimaryNetworksForTest(primaryNetworks *syncmap.SyncMap[map[string]util.NetInfo]) {
+	nadController.primaryNetworks = primaryNetworks
+}
```

Comment:
Why not have an interface for `GetActiveNetworkForNamespace` and then have a test specific implementation?

Reply:
wont that break the suggestion you gave me to use a nil struct method call?

Reply:
Well the advantage is not delivering test code to production. Bit by bit might not be much but if we keep doing it we bloat unnecessarily. It's true that sometimes it can't be avoided or the effort to do so would be unreasonable. 

`FakeNetworkController` might be another example? If that's just used in tests, shouldn't it go in `pkg/testing` somewhere?

Another potential advantage is not having people calling other current or future methods of nad controller. This is minor, I wasn't really thinking about this one.

Anyway, I don't wanna push in our urgent circumstances.

Reply:
the problem is the FakeNetworkController needs to be used by other packages testing code, and in golang you cant import test code to other test code. I guess I can use build tags and create a common package that is ignored when building ovnk, like using: // +build test

Let me know if you want me to try that.

Reply:
But that's precisely why we have `pkg/testing`, to share testing code across tests. You just import that in your tests. Like what we do with mock packages and the such.

BTW there is already a `testNetworkControllerManager` in `network_attach_def_controller_test.go`. You might find convenient to just move that one to `pkg/testing` but maybe not, your choice.

Reply:
will do that